src/main/java/org/gridsuite/securityanalysis/server/dto/ResourceFilterDTO.java

```diff
@@ -0,0 +1,65 @@
+/**
+* Copyright (c) 2023, RTE (http://www.rte-france.com)
+* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, You can obtain one at http://mozilla.org/MPL/2.0/.
+*/
+
+package org.gridsuite.securityanalysis.server.dto;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.net.URLDecoder;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+
+/**
+ * An object that can be used to filter data with the JPA Criteria API (via Spring Specification)
+ * @param dataType the type of data we want to filter (text, number)
+ * @param type the type of filter (contains, startsWith...)
+ * @param value the value of the filter
+ * @param column the column / field on which the filter will be applied
+ */
+public record ResourceFilterDTO(DataType dataType, Type type, Object value, FilterColumn column) {
+
+    public enum DataType {
+        @JsonProperty("text")
+        TEXT,
+    }
+
+    public enum Type {
+        @JsonProperty("contains")
+        CONTAINS,
+        @JsonProperty("startsWith")
+        STARTS_WITH,
+        @JsonProperty("equals")
+        EQUALS
+    }
+
+    public enum FilterColumn {
+        @JsonProperty("contingencyId")
+        CONTINGENCY_ID,
+        @JsonProperty("status")
+        STATUS,
+        @JsonProperty("subjectId")
+        SUBJECT_ID,
+        @JsonProperty("limitType")
+        LIMIT_TYPE,
+        @JsonProperty("limitName")
+        LIMIT_NAME,
+        @JsonProperty("side")
+        SIDE
+    }
+
+    public static List<ResourceFilterDTO> fromStringToList(String filters) throws JsonProcessingException {
+        if (filters == null || filters.isEmpty()) {
+            return List.of();
+        }
+        ObjectMapper objectMapper = new ObjectMapper();
```

Comment:
you can use static ObjectMapper (Note from @jonenst  in [short circuit](https://github.com/gridsuite/shortcircuit-server/pull/41#discussion_r1361691113))
ObjectMapper is thread-safe and can be reused

Reply:
I agree, it is also helpful to use the same configuration as RestTemplateConfig one

Reply:
I just used a static one...

Reply:
The DTO can't use ObjectMapper itself. The URL decoding is in the controller scope and the service will deserialize it into DTO.

Reply:
by moving the deserialisation to the service, I was able to use injected objectmapper